backward_paper/runner.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+"""
+This script contains a specialization of the Ekomark runner
+"""
+import functools
+import pathlib
+import copy
+import pandas as pd
+import numpy as np
+
+from banana import load_config
+from banana.data import dfdict
+
+from ekomark.benchmark.runner import Runner
+from ekomark import pdfname
+
+from plots import plot_pdf
+
+import lhapdf
+import eko
+
+pkg_path = pathlib.Path(__file__).absolute().parents[0]
+
+
+def rotate_to_pm_basis(log):
+    """Rotate to plus minus basis"""
+    # TODO: this can be improved
+    rot_log = {}
+    if "g" in log:
+        rot_log["g"] = log["g"]
+    for pid in eko.evolution_operator.flavors.quark_names:
+        if pid not in log:
+            continue
+        quark = log[pid]
+        qbar = log[f"{pid}bar"].copy()
+        rot_log[r"${%s}^{+}$" % (pid)] = copy.deepcopy(log[pid])
+        rot_log[r"${%s}^{-}$" % (pid)] = copy.deepcopy(log[pid])
+
+        for column_name, column_data in quark.items():
+            if column_name == "x":
+                continue
+            rot_log[r"${%s}^{+}$" % (pid)][column_name] = (
+                column_data + qbar[column_name]
+            )
+            rot_log[r"${%s}^{-}$" % (pid)][column_name] = (
+                column_data - qbar[column_name]
+            )
+    return rot_log
+
+
+class EkoRunner(Runner):
+    """
+    Specialization of the Ekomark runner.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.banana_cfg = load_config(pkg_path)
+        self.rotate_to_evolution_basis = False
+        self.plot_pdfs = [4, -4]
+        self.sandbox = True
+        self.fig_name = None
+
+    def run_external(self, theory, ocard, pdf):
+
+        if self.external == "inputpdf":
```

Comment:
Why is this not https://github.com/N3PDF/eko/blob/f04b3afa13378e9d1a4b4a376d223d2271eaf29e/benchmarks/ekomark/benchmark/runner.py#L122

Reply:
I need to modify LHAPDF utils ( loop on replicas and get `Q0` not `Q2grid`), but it's doable.

Reply:
hu? but it is at `Q2grid`:
https://github.com/N3PDF/eko/blob/f04b3afa13378e9d1a4b4a376d223d2271eaf29e/benchmarks/ekomark/benchmark/external/lhapdf_utils.py#L34
replicas - agreed, as said above ...

Reply:
Indeed, here we want to plot the pdf at the `Q0` scale not in the final `Q2grid` scale... Or am I missing the point?